pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -119,6 +116,26 @@ export async function readDeploymentProfiles(): Promise<settings.DeploymentProfi
   return profiles;
 }
 
+function readRegistryByType(regKey: nativeReg.HKEY, type: string ) {
+  let profile: undefined|RecursivePartial<settings.Settings>;
+
+  const key = nativeReg.openKey(regKey, type, nativeReg.Access.READ);
```

Comment:
After this:
```
if (!key) {
  return;
}
```

And then in the `finally` block you can just close the key without testing for it.